### ISSUE TITLE
What other /slash commands should we add?

### ISSUE THREAD

**Issue by jessepollak** [jessepollak/slash#4]
Some ideas so far:
- Youtube
- Calendar
- Amazon

**Comment by bsiddiqui** [comment]
- any doc creation/storage app (Dropbox, Quip, Google Docs) for link sharing
- trello tasks
- github pull requests - something like `/github comand/Wh` would show a link to this issue (could also be used with PRs)
- foursquare/yelp

**Comment by dget** [comment]
Two I'd love, one that's simple and one that's not:
- Some way of screenshotting another tab. Almost every PR I put on Github (any many comments) need screenshots and it's fairly cumbersome to get them in
- /shrug. Gives you ¯\_(ツ)_/¯

**Comment by jessepollak** [comment]
@dget OOOOOO I love both of these

![cat-hello-marko-xT9DPzfQYtD8lRYWs0](https://media3.giphy.com/media/xT9DPzfQYtD8lRYWs0/giphy.gif)

**Comment by heewa** [comment]
How about `/emoji cactus` for :cactus:, with the same names that github/slack/basically-everyone uses? I know you can get to the emoji from menu Edit -> Emoji & Symbols, but as slack & github show, it's a much better experience to tab-complete a word than scroll through pages of icons.

**Comment by russdogg** [comment]
Maybe something like: `/map` -> a geo-location feature that snaps a screenshot map plus google map link. Like when you need to quickly send someone your location.

![map-example](https://cloud.githubusercontent.com/assets/1434683/13473350/d15b6ace-e06d-11e5-9a8e-15cfe442e07c.jpg)

I have an IFTTT recipe that does this.

edit: looks like google maps has an API call for this, just requires long and lat and an API key
https://developers.google.com/maps/documentation/static-maps/intro

**Comment by btamayo** [comment]
:+1: for `/emoji`

Or link to Google search results for a specific term. e.g. "Great, look who's winning the primary: [2016 Primary](https://www.google.com/#q=2016+primary)"

**Comment by russdogg** [comment]
For `/emoji` , might have to add a polyfill for default emoji support for Chrome on Windows... Unless it's been updated more recently.

**Comment by jessepollak** [comment]
Love the Google Maps idea and `/emoji` is the next one I'm working on!

**Comment by jessepollak** [comment]
Just released an update (`v0.0.14`) that adds `/emoji` — it's not perfect, but it works. If anyone is around this weekend and wants to learn how to add a command, I'd be happy to help!

**Comment by dsanch3z** [comment]
@jessepollak Maybe a way to add predefined templates for emails?

Loved the extension BTW
![johnny-bravo-ohi979zEnEu4w](https://media0.giphy.com/media/ohi979zEnEu4w/giphy.gif)

**Comment by ishu3101** [comment]
@russdogg Can you share your IFTTT Recipe for Google Maps?

**Comment by ishu3101** [comment]
- Google Search - Search the web with Google
- Google Translate - Translate text using Google Translate
- Wikipedia - Search for articles on Wikipedia
- Imgur - Search for images on Imgur
- Twitter - Search tweets, @mentions and hashtags on Twitter

**Comment by chrishannah** [comment]
I've started working on the Wikipedia command! I've forked it here - https://github.com/cryptixdevelopment/command

The API is strange and returns three different arrays, each for the titles, excerpts, and urls. But I've got that converting to a more suitable array of clear objects.

It's "working" at the minute, just the results aren't formatted very well yet.

**Comment by chrishannah** [comment]
I think another good command would be to be able to quickly search on the App Store

**Comment by DanielHabib** [comment]
quickly pull up the current stock information,
`/stock <Search Stock>`

**Comment by chrishannah** [comment]
I got bored and also added in an App Store command - https://github.com/cryptixdevelopment/command

It works fine, except the pagination isn't perfect.

**Comment by jessepollak** [comment]
@cryptixdevelopment so awesome! Do you want to submit a PR? Would love to take a look and merge them in :D

**Comment by chrishannah** [comment]
@jessepollak Yeah sure, the Wiki command still needs work but at least the functionality is there.

**Comment by bgiesing** [comment]
@jessepollak I think these could be added:
- `/googleplay`
  - Android Apps OR add it to @cryptixdevelopment's command to make a `/apps` command that searches both (and possibly Windows 10, Ubuntu, etc. app stores)
- `/search`
  - Give a link to a search engine (your prefered could be set in a Settings page)
- `/lmgtfy`
  - http://lmgtfy.com/?q=let+me+google+that+for+you
- `/deviantart`
  - Share images from [DeviantArt](//deviantart.com)
- `/apkmirror`
  - Search [APK Mirror](//apkmirror.com) for Staged Rollout and Beta Android APKs
- `/producthunt`
- `/xda`
  - Search [XDA-Developers](//forum.xda-developers.com)

**Comment by scrussell24** [comment]
Awesome extension! Just thought about something similar yesterday and found this.

Have you thought about allowing custom commands? I'm thinking this could be very useful as a productivity tool in a work environment but the commands would need to be very customized and connect to internal services.

Also, the ability to "pipe" commands together would be boss. Something like "/youtube searchparam | bitly".

**Comment by ishu3101** [comment]
Custom Commands are already supported. See the CONTRIBUTING.md file here: https://github.com/jessepollak/command/blob/master/CONTRIBUTING.md to learn how to create one.

You can also see the source code of existing command such as /emoji under `src/content/commands` folder

**Comment by bgiesing** [comment]
@ishu3101 I think @scrussell24 means a easy way for users to make them with a Settings UI (so you could add macro slashes (/omw=on my way) like Slash Keyboard does on mobile). That's for devs to implement API style commands (like the YouTube API for video search for instance)

**Comment by abishekrsrikaanth** [comment]
`/stackoverflow` - Reference a Stackoverflow Page

`/github-repo` - Reference to a Github Repo
